### ISSUE TITLE
Demo of datalad's key feature when compared with Git -- seamless repo boundaries

### ISSUE THREAD

**Issue by mih** [datalad/datalad#1746]
```sh
datalad create demo
cd demo
# create subdataset
datalad create -d . sub1
# create a second subdataset nested in the previous one
datalad create -d . sub1/justadir/sub2
# now we create content in the deepest dataset
mkdir sub1/justadir/sub2/anotherdir
touch sub1/justadir/sub2/anotherdir/afile
# git only easily gives us an idea that something in
# the top-most subdataset is modified
git status
# datalad diff is more helpful
datalad diff -r
# like git it can report individual untracked files, but also
# across repository boundaries
datalad diff -r --report-untracked all

# adding content while keeping an entire dataset hierarchy in sync
# is just as easy (while in git one would need to find the right
# repository location and order first
datalad add -d . sub1/justadir/sub2/anotherdir/afile
# or just: datalad add -d. -r .
# yields a clean dataset, all affected repositories are saved
# and the associated changeset is easily discoverable from the
# super dataset
datalad diff --revision @~1 -r
```

**Comment by bpoldrack** [comment]
That's pretty cool point to make!

**Comment by mih** [comment]
This actually needs a fix to master -- coming.

**Comment by mih** [comment]
This is included in #1730 now
